File: ios/Outbound/Outbound/Core/ActivitySaveEligibility.swift
import Foundation

/// Defines when a recorded activity is substantial enough to keep as a saved activity.
nonisolated enum ActivitySaveEligibility: Equatable {
    static let minimumDurationSeconds = 300
    static let minimumDistanceMeters = 500.0

    case eligible
    case tooShort

    static func evaluate(durationSecs: Int, distanceM: Double) -> Self {
        guard durationSecs >= minimumDurationSeconds || distanceM >= minimumDistanceMeters else {
            return .tooShort
        }
        return .eligible
    }
}

File: android/core/model/src/main/kotlin/com/plainstride/outbound/core/model/activity/ActivitySaveEligibility.kt
package com.plainstride.outbound.core.model.activity

enum class SaveEligibility { eligible, tooShort }

object ActivitySaveEligibility {
    const val MINIMUM_DURATION_SECONDS = 300
    const val MINIMUM_DISTANCE_METERS = 500.0

    fun evaluate(durationSeconds: Int, distanceMeters: Double): SaveEligibility =
        if (durationSeconds >= MINIMUM_DURATION_SECONDS || distanceMeters >= MINIMUM_DISTANCE_METERS) {
            SaveEligibility.eligible
        } else {
            SaveEligibility.tooShort
        }
}
